### ISSUE TITLE
IndentationCheck: incorrect validation for class definition

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#2629]
D:> type config.xml

```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="ASCII" />
    <property name="severity" value="error" />
    <property name="fileExtensions" value="java" />

    <module name="TreeWalker">
        <module name="Indentation">
        </module>
    </module>
</module>
```

D:> type test.java

```
package test;

class
    Test1 {//indent:4 exp:0 // CS says invalid
}
public
    class Test2 {//indent:4 exp:4 // CS os OK, but this should be warning
}
```

D:> java -jar checkstyle-6.12.1-all.jar -c config.xml test.java

```
Starting audit...
test.java:4: error: 'class def ident' have incorrect indentation level 4, expected level should be 0.
Audit done.
Checkstyle ends with 1 errors.
```

I was expecting CS to not report any error on line 4 because similar code (the 2nd class) does require the line to be indented.
##

**Comment by rnveach** [comment]
D:\> type test2.java
````
package test;

class
    Test1 {//indent:4 exp:0 // CS says invalid
}
class
Test2 {//indent:0 exp:4 // CS says invalid
}
````

D:\> java -jar checkstyle-6.12.1-all.jar -c config.xml test2.java
````
Starting audit...
test2.java:4: error: 'class def ident' have incorrect indentation level 4, expected level should be 0.
test2.java:7: error: 'Test2' have incorrect indentation level 0, expected level should be 4.
Audit done.
Checkstyle ends with 2 errors.
````

I was expecting CS to not show any errors once the indentation was fixed, but this shows it can't be satisfied.

**Comment by romani** [comment]
I do not understand second example ... what you are trying to show by it ? If CS is wrong, lets describe how it should be and do not mention attempt to satisfy wrong recommendations from CS.

**Comment by rnveach** [comment]
2nd example shows that the indentation of the class definition can't be satisified by IndentationCheck.
Imagine you code only contains `Test1`. CS says it is at the wrong indentation level, column 4, and should be at column 0. You change your code based on what IndentationCheck said, to be at column 0, which is what `Test2` is showing. If you run CS again, it gives another error saying your indentation level of column 0 is wrong and it should be at column 4, which is what `Test1` was originally at. IndentationCheck says the line must be at column 0 and 4 at the same time which is impossible.

I guess it can be ignored as the correct indentation should be 4 from the first post, but I was just including it for completeness.

**Comment by DXTkastb** [comment]
I'm on it.

**Comment by DXTkastb** [comment]
> I was expecting CS to not report any error on line 4 because similar code (the 2nd class) does require the line to be indented.

@rnveach, I have a small doubt regarding this, Line no.4 should not throw an error that's true but 2nd class I guess is correctly indented, please confirm this. According to me, it is line-wrapping indentation situation and is done correctly.

**Comment by romani** [comment]
fix is merged.
